src/bootstrap/install.rs

```diff
@@ -259,10 +259,12 @@ install!((self, builder, _config),
         install_analysis(builder, self.compiler.stage, self.target);
     };
     Rustc, "src/librustc", true, only_hosts: true, {
-        builder.ensure(dist::Rustc {
-            compiler: self.compiler,
-        });
-        install_rustc(builder, self.compiler.stage, self.target);
+        for host in &builder.hosts {
```

Comment:
Is it plausible that this should happen for the other steps here? If so, we can likely achieve that by changing the macro a bit.

Currently it has:

```rust
                run.builder.ensure($name {
                    compiler: run.builder.compiler(run.builder.top_stage, run.builder.config.build),
                    target: run.target,
                });
```

For "only_hosts" steps, this will run with target = host for each host already. I think then the only difference is previously we were running the `dist::Rustc` step always with the "build" triple compiler, whereas the new code runs it with the host triple compiler. If I understand what's going on, then the code you've added could also be something like this, and without the loop, and still work.

```rust
            builder.ensure(dist::Rustc {
                compiler: builder.compiler(builder.top_stage, self.target),
            });
            install_rustc(builder, self.compiler.stage, self.target);
```

Reply:
Thanks for the alternate suggestion. I've verified this works and updated my commits.